app/containers/HomeContainer/tests/index.test.js

```diff
@@ -46,4 +57,61 @@ describe('<HomeContainer /> tests', () => {
     await timeout(500);
     expect(submitSpy).toBeCalled();
   });
+  it('should dispatch githubRepos when repoName is passed', () => {
+    const dispatch = jest.fn();
+    mapDispatchToProps(dispatch).dispatchGithubRepos({ repoName: 'Mac' });
+    expect(dispatch).toHaveBeenCalled();
+  });
+  it('should dispatch clear Repos when repoName is empty', () => {
+    const dispatch = jest.fn();
+    mapDispatchToProps(dispatch).dispatchClearGithubRepos();
+    expect(dispatch.mock.calls[0][0]).toEqual({ type: homeContainerTypes.CLEAR_GITHUB_REPOS });
+  });
+
+  it('should trigger dispatchGithubRepos on mount', async () => {
+    const getGithubReposSpy = jest.fn();
+    renderProvider(<HomeContainer dispatchGithubRepos={getGithubReposSpy} repoName="mac" />);
+    await timeout(500);
+    expect(getGithubReposSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should show repoError ', () => {
+    const { getByText } = renderProvider(<HomeContainer reposError="error" />);
+    expect(getByText('error')).toBeTruthy();
+  });
+
+  it('should set loading to true when repoName present', async () => {
+    const getGithubReposSpy = jest.fn();
+    renderProvider(<HomeContainer repoName="mac" dispatchGithubRepos={getGithubReposSpy} />);
+    await timeout(500);
+    expect(getGithubReposSpy).toHaveBeenCalledTimes(1);
+    expect(setState).toBeCalledWith(true);
+  });
+  it('should set loading to false when reposData passed', async () => {
+    renderProvider(<HomeContainer reposData={{ items: [] }} />);
+    await timeout(500);
+    expect(setState).toBeCalledWith(false);
+  });
+
+  it('should refresh page on click of button - go to story books', () => {
+    const pushSpy = jest.fn();
+    jest.spyOn(require('react-router-dom'), 'useHistory').mockReturnValue({ push: pushSpy });
+    const { getByText } = renderProvider(<HomeContainer />);
+    const clickEle = getByText('Go to Storybook');
+    fireEvent.click(clickEle);
+    expect(pushSpy).toHaveBeenCalledTimes(1);
+  });
+  it('should render  repo cards', async () => {
+    const getGithubReposSpy = jest.fn();
+    const clearGithubReposSpy = jest.fn();
+    const { getAllByTestId } = renderProvider(
+      <HomeContainer
+        dispatchClearGithubRepos={clearGithubReposSpy}
+        dispatchGithubRepos={getGithubReposSpy}
+        reposData={githubReposData}
+      />
+    );
+    await timeout(500);
+    expect(getAllByTestId('custom-card')).toHaveLength(3);
```

Comment:
yoou could do something like this 
```
expect(getAllByTestId('custom-card')).toHaveLength(githubReposData.results.length);
```

Reply:
if i've got the shape of `githubReposData` right, then `githubReposData.results.length` should evaluate to 3 as well. Just makes sure that 
1. my tests don't break if I change the value of `githubReposData`
2. Becomes easier to read for another developer in the team